rpd_generator/bdl_structure/bdl_commands/zone.py

```diff
@@ -320,6 +320,52 @@ def populate_data_elements(self):
                     zone_ef_power_per_flow * exhaust_airflow
                 )
             return
+        # Populate Terminal for terminals_has_demand_control_ventilation
+        if self.keyword_value_pairs.get(BDL_ZoneKeywords.OA_FLOW_PER) is not None:
+            if (
+                self.parent.keyword_value_pairs.get(BDL_SystemKeywords.DOA_SYSTEM)
+                is not None
+            ):
+                doas = self.parent.keyword_value_pairs.get(
+                    BDL_SystemKeywords.DOA_SYSTEM
+                )
+                if (
+                    self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == "DCV-ZONE-SENSORS"
+                ):
+                    self.terminals_has_demand_control_ventilation = [False, False, True]
+                else:
+                    if (
+                        self.parent.keyword_value_pairs.get(
+                            BDL_SystemKeywords.DOAS_ATTACHED_TO
+                        )
+                        == "AHU-MIXED-AIR"
+                        and self.parent.keyword_value_pairs.get(
+                            BDL_SystemKeywords.MIN_OA_METHOD
+                        )
+                        == "DCV-RETURN-SENSORS"
+                    ):
+                        self.terminals_has_demand_control_ventilation = [
+                            False,
+                            False,
+                            True,
+                        ]
+            else:
+                if (
+                    self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == "DCV-ZONE-SENSORS"
+                    or self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == "DCV-RETURN-SENSOR"
+                ):
+                    self.terminals_has_demand_control_ventilation = [True, False, False]
```

Comment:
same Q about all 3 terminal types vs. only at position [0]

Reply:
@JacksonJ-KC See response above. In all cases we can safety set all three.

Reply:
The code treats False differently then None. We should only populate data associated with things that we know exist. If there is no DOAS terminal and no baseboard terminal the data needs to stay as None - it can't be False

Reply:
Addressed